Nodejs/Product/TestAdapter/TestExecutorWorker.cs

```diff
@@ -316,11 +318,47 @@ void RecordEnd(TestCaseResult test, ResultObject resultObject)
             }
         }
 
+        private int GetParentProcessId(uint processId, int counter = 0)
+        {
+            var query = string.Format("SELECT ParentProcessId FROM Win32_Process WHERE ProcessId = {0}", processId);
+            var search = new ManagementObjectSearcher("root\\CIMV2", query);
+            var results = search.Get().GetEnumerator();
+            results.MoveNext();
+
+            var parentId = (uint)results.Current["ParentProcessId"];
+
+            try
+            { 
+                var parentName = Process.GetProcessById((int)parentId).ProcessName;
+
+                if (parentName == "devenv")
+                    return (int)parentId;
+                if (counter < 8) //devenv.exe is usually the 4th process app, so don't search forever
+                    return GetParentProcessId(parentId, counter++);
```

Comment:
nit: I prefer to use a for loop as I feel is more readable. Feel free to ignore this anyway is just preference.

Reply:
I feel you, totally, but I still think this is cleaner.